### PR TITLE
extractor: added vidplay and fix few extractors

### PR DESCRIPTION
**Pull request by Sofie99**

### CODE REVIEW

**Comment by b4byhuey** [comment]
Can you fix Dailymotion? or can you conver from aniyomi [here](https://github.com/aniyomiorg/aniyomi-extensions/raw/master/lib/dailymotion-extractor/src/main/java/eu/kanade/tachiyomi/lib/dailymotionextractor/DailymotionExtractor.kt)

File: app/src/main/java/com/lagradost/cloudstream3/extractors/Vidplay.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                this.name,
+                it.file ?: return@map,
+                "$mainUrl/"
+            ).forEach(callback)
+        }
+
+    }
+
+    private suspend fun getKeys(): List<String> {
+        return app.get("https://raw.githubusercontent.com/Claudemirovsky/worstsource-keys/keys/keys.json")
```
Comment (fire-light42) [inline app/src/main/java/com/lagradost/cloudstream3/extractors/Vidplay.kt:47]: this should be cached in a local variable

**Review by fire-light42 — CHANGES_REQUESTED** [review]
We dont want to ddos github for every call to geturl, save it to a local variable even if app.get may be cached it does not hurt to explicitly cache it

**Review by fire-light42 — APPROVED** [review]

**Comment by fire-light42** [comment]
Understandable, I can merge this now if you want

**Pull request merged**